### PR TITLE
chore: Increase unit test code coverage of Xml FieldReader and Xml FieldWriter

### PR DESCRIPTION
**Pull request by joeylimd**
Relates to: #203

### CODE REVIEW

File: runtime/modules/xml/core/src/test/java/io/atlasmap/xml/core/XmlFieldReaderTest.java [hunk 1]
```diff
@@ -359,4 +366,673 @@ private String getDocumentString(String uri) throws IOException {
         fis.close();
         return new String(buf);
     }
+
+    @Test
+    public void testXmlFieldDoubleMax() throws Exception {
+        String fieldPath = "/primitive/value";
```
Comment (rohanmars) [inline runtime/modules/xml/core/src/test/java/io/atlasmap/xml/core/XmlFieldReaderTest.java]: Can we make some private/protected methods for repetitious code? There seems to a few standard code patterns in use in the file. Make some methods which parametrize differing values.

Comment (joeylimd) [reply runtime/modules/xml/core/src/test/java/io/atlasmap/xml/core/XmlFieldReaderTest.java]: Done. extracted private method for repetitive codes. Please review. Thanks.

**Comment by rohanmars** [comment]
LGTM :+1:

**Comment by rohanmars** [comment]
Joey, changes look good. I wonder if it's worth adding some tests to validate negative scenarios of the read and s etDocument methods. For example, if incorrect xml is passed. The idea would be test situations which should generate exceptions. What do you think?

**Comment by rohanmars** [comment]
@joeylimd , the changes look good and meet the requirements in @203.

LGTM :+1:
